File: src/test/java/com/tesseractservers/BitFlagsTest.java
package com.tesseractservers;

import org.junit.Test;

import java.util.Random;

public class BitFlagsTest {

	public static final int ITERATIONS = 4096;

	private long randomFlag(Random random) {
		return (long) Math.pow(2, random.nextInt(63));
	}

	@Test
	public void testFlagSetting() {
		Random random = new Random();
		long flags = 0;

		for (int i = 0; i < ITERATIONS; i++) {

			long flag = randomFlag(random);
			flags = BitFlags.setFlag(flags, flag);
			assert BitFlags.hasFlag(flags, flag);

			if (random.nextBoolean()) {
				flags = BitFlags.unsetFlag(flags, flag);
				assert !BitFlags.hasFlag(flags, flag);
			}

		}
	}

}

File: src/test/java/net/proxzi/BitFlagsTest.kt
package net.proxzi

import net.proxzi.BitFlags.hasFlag
import net.proxzi.BitFlags.setFlag
import net.proxzi.BitFlags.unsetFlag
import org.junit.Test
import java.util.*
import kotlin.math.pow

class BitFlagsTest {

	private fun randomFlag(random: Random): Long {
		return 2.0.pow(random.nextInt(63).toDouble()).toLong()
	}

	@Test
	fun testFlagSetting() {
		val random = Random()
		var flags: Long = 0
		for (i in 0 until ITERATIONS_FAST) {
			val flag = randomFlag(random)
			flags = setFlag(flags, flag)
			assert(hasFlag(flags, flag))
			if (random.nextBoolean()) {
				flags = unsetFlag(flags, flag)
				assert(!hasFlag(flags, flag))
			}
		}
	}

}
